RealmSwift/Tests/ListTests.swift

```diff
@@ -643,3 +643,103 @@ class ListRetrievedTests: ListTests {
         return array
     }
 }
+
+#if swift(>=4.0)
```

Comment:
Any reason not to run these tests on older versions of Swift? Even if they're not actually exercising our code, they should work and would serve as a sanity check.

Reply:
Yeah, running them for all versions makes sense.